src/composables/map/street-view.composable.ts

```diff
@@ -0,0 +1,257 @@
+import { watch, nextTick } from 'vue'
+import { storeToRefs } from 'pinia'
+import useMap from '@/composables/map/map.composable'
+import { useAppStore } from '@/stores/app.store'
+import { useInfoStore } from '@/stores/info.store'
+import { MapBrowserEvent } from 'ol'
+import BaseEvent from 'ol/events/Event'
+import { Coordinate } from 'ol/coordinate'
+import { Select } from 'ol/interaction'
+import { fromLonLat, toLonLat } from 'ol/proj'
+import { containsCoordinate } from 'ol/extent.js'
+import { loadGoogleapis } from '@/services/info/street-view'
+import { SvCompassFeature } from '@/services/info/sv-compass-feature'
+import { SvDirectionFeature } from '@/services/info/sv-direction-feature'
+
+import { Feature } from 'ol'
+import { Point } from 'ol/geom'
+import VectorLayer from 'ol/layer/Vector'
+import VectorSource from 'ol/source/Vector'
+
+export const DEFAULT_INFO_ZINDEX = 1500
+export const SV_FEATURE_LAYER_TYPE = 'svFeatureLayer'
+
+const SV_RADIUS = 90
+
+export default function useStreeView() {
+  const { infoOpen } = storeToRefs(useAppStore())
+  const {
+    locationInfo,
+    hidePointer,
+    isStreetviewActive,
+    noDataAtLocation,
+    panoPositionChanging,
+    svFeature,
+  } = storeToRefs(useInfoStore())
+
+  const map = useMap().getOlMap()
+
+  let googleService: any = null
```

Comment:
Please add typing as much as possible.

Reply:
I don't have any type for the global google service in the window, I find it hacky anyway to use the window scope, but this work with code migrated from v3